apps/web/src/composables/contracts.ts

```diff
@@ -426,6 +164,165 @@ export default function useContracts() {
+        return feesRounded
+    }
+
+    async function _getPools(operatorId: number): Promise<Pool[]> {
+        const pools: Pool[] = []
+
+        const poolIds = [
+            ...await manager.getPendingPoolIds(),
+            ...await manager.getStakedPoolIds()
+        ]
+    
+        for (const poolId of poolIds) {
+            const poolDetails = await casimirViews.getPoolDetails(poolId)
+            const pool = {
+                ...poolDetails,
+                operatorIds: poolDetails.operatorIds.map(id => id.toNumber()),
+                reshares: poolDetails.reshares.toNumber()
+            }
+            if (pool.operatorIds.includes(operatorId)) {
+                pools.push(pool)
+            }
+        }
+        return pools
+    }
+
+    async function getUserOperators(): Promise<void> {
+        const userAddresses = (user.value as UserWithAccountsAndOperators).accounts.map((account: Account) => account.address) as string[]
+
+        const scanner = new Scanner({ 
+            ethereumUrl,
+            ssvNetworkAddress,
+            ssvNetworkViewsAddress
+        })
+
+        const ssvOperators: Operator[] = []
+        for (const address of userAddresses) {
+            const userOperators = await scanner.getOperators(address)
+            ssvOperators.push(...userOperators)
+        }
+
+        const casimirOperators: RegisteredOperator[] = []
+        for (const operator of ssvOperators) {
+            const { active, collateral, poolCount, resharing } = await casimirOperatorRegistry.getOperator(operator.id)
+            const registered = active || collateral.gt(0) || poolCount.gt(0) || resharing
+            if (registered) {
+                const pools = await _getPools(operator.id)
+                // TODO: Replace once we have this working again
+                const operatorStore = {
+                    '654': 'https://nodes.casimir.co/eth/goerli/dkg/1',
+                    '655': 'https://nodes.casimir.co/eth/goerli/dkg/2',
+                    '656': 'https://nodes.casimir.co/eth/goerli/dkg/3',
+                    '657': 'https://nodes.casimir.co/eth/goerli/dkg/4',
+                    '658': 'https://nodes.casimir.co/eth/goerli/dkg/5',
+                    '659': 'https://nodes.casimir.co/eth/goerli/dkg/6',
+                    '660': 'https://nodes.casimir.co/eth/goerli/dkg/7',
+                    '661': 'https://nodes.casimir.co/eth/goerli/dkg/8'
+                }
+                const url = operatorStore[operator.id.toString() as keyof typeof operatorStore]
+                casimirOperators.push({
+                    ...operator,
+                    active,
+                    collateral: ethers.utils.formatEther(collateral),
+                    poolCount: poolCount.toNumber(),
+                    url,
+                    resharing,
+                    pools
+                })
+            }
+        }
+        _setOperators(ssvOperators)
```

Comment:
What if we only push non-Casimir-registered operators to unregisteredOperators and Casimir-registered operators to registeredOperators and join the lists if needed (but removes some other diffing logic elsewhere to have them separate).

Reply:
💡